src/app/static/src/app/components/projects/ProjectHistory.vue

```diff
@@ -313,6 +413,38 @@
                     this.renamedProjectName = "";
                 }
             },
+            cancelNoteEditing() {
+                if (this.versionNoteToEdit) {
+                    this.versionNoteToEdit = null;
+                    this.editedNote = "";
+                }
+
+                if (this.projectNoteToEdit) {
+                    this.projectNoteToEdit = null
+                    this.editedNote = "";
+                }
+            },
+            async confirmNoteEditing() {
+                if (this.versionNoteToEdit) {
+                    const versionPayload: versionPayload = {
+                        version: this.versionNoteToEdit!,
+                        note: this.editedNote
+                    };
+                    this.updateVersionNoteAction(versionPayload);
+                    this.versionNoteToEdit = null;
+                    this.editedNote = "";
+                }
+
+                if (this.projectNoteToEdit) {
```

Comment:
Could make this an `else if` just to emphasise to the reader that it ought to be one or the other.